Source/Test/Rewriting/Passes/AssemblyRewriter.cs

```diff
@@ -632,5 +637,23 @@ protected static string GetFullyQualifiedMethodName(MethodReference method)
 
             return name;
         }
+
+        /// <summary>
+        /// Replaces the existing instruction with a new instruction and fixes up any
+        /// branch references to the old instruction so they point to the new instruction.
+        /// </summary>
+        /// <param name="instruction">The instruction to be replaced.</param>
+        /// <param name="newInstruction">The new instruction to be inserted.</param>
+        protected void Replace(Instruction instruction, Instruction newInstruction)
+        {
+            this.Processor.Replace(instruction, newInstruction);
+            foreach (var i in this.Processor.Body.Instructions)
+            {
+                if (i.Operand is Instruction j && j == instruction)
+                {
+                    i.Operand = newInstruction;
+                }
+            }
+        }
```

Comment:
Would it make sense to set `ModifiedMethodBody` to `true` here whenever we invoke this method, as we already do it explicitly?